File: leetcode/3169.js
/**
 * @param {number} days
 * @param {number[][]} meetings
 * @return {number}
 */
var countDays = function(days, meetings) {
    meetings.sort((s, e) => s[0] - e[0]);

    for (let i=1; i<meetings.length; i++) {
        let [start, end] = meetings[i];
        let [prevStart, prevEnd] = meetings[i - 1];

        if (start <= prevEnd) {
            meetings[i][0] = prevStart;
            meetings[i][1] = Math.max(end, prevEnd);
        }

        if (prevStart !== meetings[i][0]) {
            days -= (prevEnd - prevStart + 1);
        }
    }
    answer = days - (meetings[meetings.length - 1][1] - meetings[meetings.length - 1][0] + 1);
    return answer
};

File: leetcode/3169.py
# https://leetcode.com/problems/count-days-without-meetings/description/
# 2025-03-27
# aystudy

class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        flag = [1] * (days + 1)

        for i in range(len(meetings)):
            for j in range(meetings[i][0], meetings[i][1] + 1): 
                flag[j] = 0
        
        answer = sum(flag) - 1
        return answer